src/interval.rs

```diff
@@ -0,0 +1,785 @@
+use std::ops::{Add, Div, Mul, Sub};
+
+use num_traits::{Bounded, FromPrimitive, ToPrimitive};
+
+/// Interval is a set of real numbers lying between two numbers, the extremities
+/// (left and right) of the interval.
+/// # Source
+/// https://en.wikipedia.org/wiki/Interval_(mathematics)
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Interval<T> {
+    left: T,
+    right: T,
+    left_infinite: bool,
+    right_infinite: bool,
```

Comment:
I'm considering a refactor: instead of storing these two flags, this could be handled by the constructors - if the interval should be infinite, then the infinite side can be set to the `min_value` or `max_value` of `T`. Because `T` it implements `Bounded` trait, there always are specific numeric extremes. 
It would make the implementation simpler (no infinity checks anymore) and it also makes an entirely infinite (left and right) interval possible. 
It will be then also possible to remap a finite interval into an infinite interval, which might be mathematically incorrect but it could be quite handy in our case. 
We currently use the Interval only to specify a value range of voxelized scalar field, that should be considered to be a volume. The scalar field can be constructed from a mesh - imagine a sphere. The voxels representing the sphere surface will be set to value **0** (this is customizable) and the rest will be set to None. Converting interval **0to0** into a mesh materializes the voxelized sphere surface and generates two surfaces - outer and inner. Then it's possible to compute the distance field. Each voxel will be set a value representing its distance from the original mesh surface. The voxels that used to be **0** will remain **0**, the ones next tot them will be **1**, next layer will be **2** etc. But not only that - the voxels inside closed mesh surfaces will be set to negative values, so the sphere surface will be **0**, the concentric sphere bigger than the original will be set to **1** and the concentric sphere smaller than the original will be set to **-1** etc. Now it's possible to materialize mesh from **0to0**, which will be the same as before. Interval **-1to1** will be 3 voxels thick. Interval **infinityTo0** will be the sphere including its inside, so there won't be any inner surface materialized into mesh. Similarly, **1toInfinity** is the entire outer space.
When performing boolean or other operations on two scalar fields, it is valid to make i.e. an union of field voxels within interval **-1to1** (thicker hollow surface) and **infinityTo0** (filled volumes) and remapping values from **infinityTo0** to **-1to1**. The result will be valid and most probably will look good. 
To the best visual outputs, it would be best to calculate the actual value of infinity (the lowest or highest actually used value) and set the numerical values of interval extremes accordingly.